overlord/snapstate/autorefresh.go

```diff
@@ -526,3 +550,17 @@ func inhibitRefresh(st *state.State, snapst *SnapState, info *snap.Info, checker
 	}
 	return nil
 }
+
+func (m *autoRefresh) isRefreshHeld(refreshSchedule []*timeutil.Schedule) (bool, time.Time, error) {
```

Comment:
I found the name somewhat odd: what does it mean to hold a refresh? To be in progress or to block it? Either reading is plausible based on the method name alone. Perhaps add a docstring to the function to explain that (some IDEs show things like that).

```
// isRefreshHeld returns true if a refresh hold is in place and refreshes are blocked
//
// In addition, if true, the deadline for the hold is returned
```